src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,67 @@
+import { type ChangeEvent, useCallback, useEffect, useState } from 'react';
+import { cn } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+  customSlippageEnabled,
+}: SwapSettingsSlippageInputReact) {
+  const { maxSlippage, setMaxSlippage } = useSwapContext();
+  const [slippageValue, setSlippageValue] = useState(maxSlippage.toString());
+
+  useEffect(() => {
+    if (customSlippageEnabled) {
```

Comment:
Add some comment here to help understand what's going on

Reply:
Will do